src/core/messaging/ACSClient.ts

```diff
@@ -219,7 +219,7 @@ export class ACSConversation {
                 }
 
                 // Filter out duplicate messages
-                if (postedMessageIds.has(id)) {
+                if (postedMessageIds.has(id) && postedMessageIds.has(version)) {

```

Comment:
Are we certain `version` will always take the same format as `id`? Is it safer if we use something like `${id}_version_${version}` inside the `postedMessageIds` set?

Reply:
After discussion with @xTEddie , updated logic to rely on `editedOn` property instead of `version`.